json-logger/src/main/java/org/mule/extension/jsonlogger/internal/datamask/JsonMasker.java

```diff
@@ -1,133 +1,90 @@
 package org.mule.extension.jsonlogger.internal.datamask;
 
-import com.fasterxml.jackson.annotation.JsonInclude;
-import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.JsonNode;
-import com.fasterxml.jackson.databind.MapperFeature;
-import com.fasterxml.jackson.databind.ObjectMapper;
-import com.fasterxml.jackson.databind.SerializationFeature;
 import com.fasterxml.jackson.databind.node.ArrayNode;
 import com.fasterxml.jackson.databind.node.ObjectNode;
 import com.fasterxml.jackson.databind.node.TextNode;
-import com.jayway.jsonpath.Configuration;
-import com.jayway.jsonpath.JsonPath;
-import com.jayway.jsonpath.Option;
-import com.jayway.jsonpath.spi.json.JacksonJsonNodeJsonProvider;
 
 import java.util.*;
 import java.util.regex.Pattern;
 
 public class JsonMasker {
-
-    private static final Pattern digits = Pattern.compile("\\d");
-    private static final Pattern capitalLetters = Pattern.compile("[A-Z]");
-    private static final Pattern nonSpecialCharacters = Pattern.compile("[^X\\s!-/:-@\\[-`{-~]");
-
-    private static final Configuration jsonPathConfig = Configuration.builder()
-            .jsonProvider(new JacksonJsonNodeJsonProvider())
-            .options(Option.AS_PATH_LIST, Option.SUPPRESS_EXCEPTIONS).build();
-
-    private final Set<String> blacklistedKeys;
-    private final Set<JsonPath> blacklistedJsonPaths;
-    private final boolean enabled;
-
-    public JsonMasker(Collection<String> blacklist, boolean enabled) {
-        this.enabled = enabled;
-
-        blacklistedKeys = new HashSet<>();
-        blacklistedJsonPaths = new HashSet<>();
```

Comment:
No need for a path check. For one it is case sensitive, and secondly the same task can be accomplished using just the keys.